src/logging.h

```diff
@@ -157,6 +157,14 @@ namespace BCLog {
         /** Only for testing */
         void DisconnectTestLogger() EXCLUSIVE_LOCKS_REQUIRED(!m_cs);
 
+        /** Disable logging
+         * This offers a slight speedup and slightly smaller memory usage
+         * compared to leaving the logging system in its default state.
+         * Mostly intended for libbitcoin-kernel apps that don't want any logging.
+         * Should be used instead of StartLogging().
+         */
+        void DisableLogging() EXCLUSIVE_LOCKS_REQUIRED(!m_cs);
```

Comment:
In commit "logging: Add DisableLogging()" (0b1960f1b29cfe5209ac68102c8643fc9553f247)

I think it's a good idea to add this `DisableLogging()` method for clarity, even though in practice it's equivalent to just calling `StartLogging()` which setting any logging options.

But I think in the future it would be nice to simplify the log interface and replace `StartLogging()` `DisableLogging()` and `DisconnectTestLogger()` methods with a single SetOptions() method:

```c++
struct BufferOptions {
   size_t max_bytes{1000000};
};

struct OutputOptions {
    fs::path print_to_file{};
    bool print_to_console{false};
    bool log_timestamps{true};
    bool log_time_micros{false};
    bool log_threadnames{false};
    bool log_sourcelocations{false};
    bool always_print_category_level{false};
};

using Options = std::variant<BufferOptions, OutputOptions>;

Options m_options{};

void SetOptions(Options options);
```

The default behavior, as currently, would be to buffer log messages, but it would be easy to set default output options to disable logging, or set custom output options to enable it.

Reply:
I'd like to change `Enabled()` and `WillLogCategoryLevel()` functions to be lockless so that calls to `Log*()` don't block when they have nothing to do. I think the above changes would mesh well with that.